nightfall-optimist/src/services/block-assembler.mjs

```diff
@@ -116,46 +125,56 @@ export async function conditionalMakeBlock(proposer) {
         });
 
         // propose this block to the Shield contract here
-        const unsignedProposeBlockTransaction = await (
-          await waitForContract(STATE_CONTRACT_NAME)
-        ).methods
+        const txDataToSign = await stateContractInstance.methods
           .proposeBlock(
             Block.buildSolidityStruct(block),
             transactions.map(t => Transaction.buildSolidityStruct(t)),
           )
           .encodeABI();
+        const { address } = web3.eth.accounts.privateKeyToAccount(ethPrivateKey);
+        const blockStake = await stateContractInstance.methods.getBlockStake().call();
+        const tx = {
+          from: address,
+          to: stateContractAddress,
+          data: txDataToSign,
+          value: blockStake,
+          gas: 8000000,
+        };
+        const signedTx = await web3.eth.accounts.signTransaction(tx, ethPrivateKey);
```

Comment:
i know you mention this on the description, but final version must ensure that multiple L2 blocks can fit in the same L1 block.

Reply:
Noted :memo: Thanks for taking a look